docs/rfc/rfc-1.adoc

```diff
@@ -633,13 +633,138 @@ As time passes and operators drop out of the system, a wallet becomes at risk
+  before it is at risk of punishment.
+- `wallet_max_age`: The oldest we allow a wallet to become before we transfer the funds
+  to a randomly selected wallet.
+- `wallet_min_btc`: The smallest amount of btc a wallet can hold before we
+  attempt to close the wallet and transfer the funds to a randomly selected
+  wallet.
+
+When a wallet fails a <<heartbeat,heartbeat>> `consecutive_failed_heartbeats`
+times, ends a <<redemption,redemption>> with less than `wallet_min_btc`
+remaining, or exceeds the `wallet_max_age`, the operators begin the process of
+closing it. If there are still funds remaining, the <<operator-only,first
+operator>> posts a <<operator-only,reimbursable>> transaction to the ethereum
+chain declaring an intention to close the wallet.
+
+That operator proposes a BTC fee (as in <<sweeping,sweeping>>) and the
+operators use the wallet's public key hash to choose viable (it must also have
+a passing heartbeat and be above `wallet_min_btc`, and not exceeded the
+`wallet_max_age) wallet(s) to transfer the remaining funds to. They construct a
+P2PKH transaction moving the wallet's main UTXO to that wallet. For more
+details on exactly how a wallet is chosen and how the funds are split up see
+<<wallet-closure-transaction,the dedicated section>>.
+
+After this transaction is complete, <<public-knowledge,the public>> is able to
+submit a SPV proof to let ethereum know that the funds were transferred
+nonfraudulently. Later, a <<donate,donation>> can be made by governance to
+handle any difference in outstanding v2 token supply and locked BTC due to
+mining transfer fees.
+
+*Notes*: Transferring the BTC to any address other than the P2PKH of one of the
+other wallet addresses is fraud, and is <<Punishment,punishable>>. Moving funds
+without sending a follow-up proof is <<Punishment,punishable>>. Failure to
+close a wallet that failed a <<heartbeat,heartbeat>> or fell below
+`wallet_min_btc`, or exceeded `wallet_max_age` is <<Punishment,punishable>>.
+Failure to complete the process before `wallet_closure_timeout` has elapsed is
+<<Punishment,punishable>>.
+
+Any unswept funds can be returned via the 30-day return script, though since
+we're redeeming from the oldest wallet and only sweeping/depositing to the
+newest wallets this hopefully won't come up often 🤞.
+
+[[wallet-closure-transaction]]
+===== Wallet Closure Transactions
+
+Governable Parameters:
+
+- `wallet_transfer_max`: The most amount of BTC a wallet can transfer to a
+  single other wallet during a wallet closure.
+
+When a wallet <<closing-a-wallet,closes>>, if there is any amount of bitcoin
+remaining, it needs to be transfered to another _live_ wallet. Our input to the
+transaction is the closing wallet's main UTXO, and we create an output for
+every multiple of `wallet_transfer_max` we have in BTC in the closing wallet.
+
+The destination wallets are chosen for their public key hash's modulus distance
```

Comment:
If you don't do the modulus then numbers in the middle of the range are statistically more likely to be chosen, since they're closer to more numbers than numbers at the ends. Here's a simulated output:

```
const arr = [0,1,2,3,4,5,6,7]
let results = {}
for (let i = 0; i < 1000; i++) {
  const sample = getRandomSample(arr, 3)
  const option1DistanceSquared = Math.pow(sample[0] - sample[1], 2)
  const option2DistanceSquared = Math.pow(sample[0] - sample[2], 2)
  if (option1DistanceSquared < option2DistanceSquared) {
    results[sample[1]] = (results[sample[1]] || 0) + 1
  } else if (option1DistanceSquared > option2DistanceSquared) {
    results[sample[2]] = (results[sample[2]] || 0) + 1
  } else if (sample[1] > sample[2]){
    results[sample[1]] = (results[sample[1]] || 0) + 1
  } else {
    results[sample[2]] = (results[sample[2]] || 0) + 1
  }
}

console.log(results)
```
`getRandomSample` is grabbing a random sample from `[0,1,2,3,4,5,6,7]` like `[4,2,6]`. The first number is the wallet that's being closed and the second and third are the options. Here's how often each number is chosen without a modulus from 1000 runs:

```
{
  '0': 48,
  '1': 108,
  '2': 154,
  '3': 163,
  '4': 151,
  '5': 147,
  '6': 152,
  '7': 77
}
```

But with a modulus:

```
function dist(a, b, modulus) {
  let amod = a % modulus
  let bmod = b % modulus

  if (amod < bmod) {
    const temp = amod
    amod = bmod
    bmod = temp
  } 
  return Math.min(amod-bmod, bmod-amod+modulus);
}


const arr = [0,1,2,3,4,5,6,7]
let results = {}
for (let i = 0; i < 1000; i++) {
  const sample = getRandomSample(arr, 3)
  const option1Distance = dist(sample[0], sample[1], arr.length)
  const option2Distance = dist(sample[0], sample[2], arr.length)

  if (option1Distance < option2Distance) {
    results[sample[1]] = (results[sample[1]] || 0) + 1
  } else if (option1Distance > option2Distance) {
    results[sample[2]] = (results[sample[2]] || 0) + 1
  } else if (sample[1] > sample[2]){
    results[sample[1]] = (results[sample[1]] || 0) + 1
  } else {
    results[sample[2]] = (results[sample[2]] || 0) + 1
  }
}
console.log(results)
```

We get a nice random distribution
```
{
  '0': 122,
  '1': 111,
  '2': 120,
  '3': 105,
  '4': 119,
  '5': 128,
  '6': 159,
  '7': 136
}
```